test/integrationtests/voight_kampff/test_setup.py

```diff
@@ -0,0 +1,163 @@
+
+
+def main(cmdline_args):
+    """Parse arguments and run environment setup."""
+    platforms = list(MycroftSkillsManager.SKILL_GROUPS)
+    parser = argparse.ArgumentParser(formatter_class=RawTextHelpFormatter)
+    parser.add_argument('-p', '--platform', choices=platforms)
+    parser.add_argument('-t', '--test-skills', default=[],
+                        help=('Comma-separated list of skills to test.\n'
+                              'Ex: "mycroft-weather, mycroft-stock"'))
+    parser.add_argument('-s', '--extra-skills', default=[],
+                        help=('Comma-separated list of extra skills to '
+                              'install.\n'
+                              'Ex: "cocktails, laugh"'))
+    parser.add_argument('-r', '--random-skills', default=0,
+                        help='Number of random skills to install.')
+    parser.add_argument('-d', '--skills-dir')
+    parser.add_argument('-c', '--config',
+                        help='Path to test configuration file.')
+
+    args = parser.parse_args(cmdline_args)
+    if args.test_skills:
+        args.test_skills = args.test_skills.replace(',', ' ').split()
+    if args.extra_skills:
+        args.extra_skills = args.extra_skills.replace(',', ' ').split()
+
+    if args.config:
+        load_config(args.config, args)
+
+    if args.platform is None:
+        args.platform = "mycroft_mark_1"
+
+    msm = MycroftSkillsManager(args.platform, args.skills_dir)
+    run_setup(msm, args.test_skills, args.extra_skills, args.random_skills)
+
+
+def run_setup(msm, test_skills, extra_skills, num_random_skills):
+    """Install needed skills and collect feature files for the test."""
+    skills = [msm.find_skill(s) for s in test_skills + extra_skills]
+    # Install test skills
+    for s in skills:
+        if not s.is_local:
+            print('Installing {}'.format(s))
+            msm.install(s)
+        else:
+            try:
+                msm.update(s)
+            except MsmException:
+                pass
+
+    # collect feature files
+    for skill_name in test_skills:
+        skill = msm.find_skill(skill_name)
+        behave_dir = join(skill.path, 'test', 'behave')
+        if Path(behave_dir).exists():
+            copy_feature_files(behave_dir, FEATURE_DIR)
+
+            step_dir = join(behave_dir, 'steps')
+            if Path().exists():
+                copy_step_files(step_dir, join(FEATURE_DIR, 'steps'))
+        else:
+            # Generate feature file if no data exists yet
+            print('No feature files exists for {}, '
+                  'generating...'.format(skill_name))
+            # No feature files setup, generate automatically
+            feature = generate_feature(skill_name, skill.path)
+            with open(join(FEATURE_DIR, skill_name + '.feature'), 'w') as f:
+                f.write(feature)
+
+    # Install random skills from uninstalled skill list
+    random_skills = [s for s in msm.all_skills if s not in msm.local_skills]
```

Comment:
Shouldn't this be optional?  We don't want every test run to have random skills, just those that run outside of pull requests.

Reply:
This only get's the skill-list, lines below selects the actual random skills from the list of non-local skills.